wordpress-mobile/WordPress-iOS
# Add navigation to the new Me Scene from Spotlight Search and Universal links.

#### Replaces the existing calls to the `Me` tab with the new modal presentation, from Spotlight Search and Universal links. Also adds the Me button to My Sites if no site is available, and removes the `Me` tab from the app.

Ref #13319

#### Change log
- Add `WPTabBarController+MeNavigation.swift`, replace the old calls with new calls to the new `Me` scene
- Update `MeViewController.swift`, add method `refreshDetailView()` that fixes a layout bug when rotating an iPhone from compact to regular
- Update `BlogDetailsViewController+MeBarButton.swift`, add accessibility to the `Me` gravatar button
- Update BlogListViewController+MeBarButton.swift, renamed to UIBarButtonItem+MeBarButton.swift as now contains extension from blog list and blog details
- Update BlogListViewController.h, added scenePresenter property
- Update BlogListViewController.m, added logic in setAddSiteBarButtonItem to add the Me button if no sites are available
- Add unit tests for the `Me` gravatar button in `My Site` (`BlogDetailsViewController+MeButtonTests.swift`)
- Fixed crash in unit tests
- Rename `scenePresenter` to `meScenePresenter` in `BlogLis`t, `BlogDetail` and `WPtabController`
- Improved dependency injection of `meScenePresenter` in `BlogList` and `BlogDetails`
- Update `WPTabBarController.m`, remove the `Me` tab
- As discussed with @bjtitus , update `NewBlogDetailHeaderView.swift`, make `delegate` a `weak var` to prevent leaks
- Update UI tests to reflect the new UI

To test:

Universal links
- With the app installed on the simulator, open terminal and run any of the following commands
  - `xcrun simctl openurl booted https://wordpress.com/me`
  - `xcrun simctl openurl booted https://wordpress.com/me/account`
  - verify that the corresponding screen opens (Me or Account Settings)

Spotlight
- From the home screen, swipe down to show the Spotlight Search bar
- Search for WordPress
- In the search results for the WordPress app, tap `Me` and verify that the new `Me` scene opens
- If `Me` doesn't show, open the app and access the screen a few times so it gets indexed. You can do the same with `Me->App Settings` and all the other sub levels.

No Sites
- Log in with an account that has no sites
- Verify that the `Me` button appears in the top right and works as expected.

Me tab removed
- Ensure that the `Me` tab does not appear in the tab bar
- Tap on the various tabs and verify that they behave as expected
- Trigger the tab navigation remotely, using the commands below and verify that they show the corresponding tab
  - `xcrun simctl openurl booted https://wordpress.com/read` (reader)
  - `xcrun simctl openurl booted https://wordpress.com/notifications` (notifications)



PR submission checklist:
